tests/main/views/test_direct_award.py

```diff
@@ -934,22 +935,296 @@ def test_record_project_download_creates_audit_event(self):
+            "has_active_search",
+            "has_non_active_search",
+            "active_search_has_chosen_sid",
+            "non_active_search_has_chosen_sid",
+            "project_locked",
+            "existing_process_outcome_result",
+            "existing_process_outcome_complete",
+            "expected_status_code",
+            "expected_response_data",
+        ),
+        (
+            # "happy" paths
+            (True, True, True, True, True, None, False, 200, AnySupersetOf({}),),
+            (True, False, True, False, True, None, False, 200, AnySupersetOf({}),),
+            (True, False, True, False, True, "awarded", False, 200, AnySupersetOf({}),),
+            (True, True, True, False, True, "cancelled", False, 200, AnySupersetOf({}),),
+            # "failure" paths
+            (
+                True,
+                True,
+                False,
+                True,
+                True,
+                None,
+                False,
+                404,
+                {
+                    "error": AnyStringMatching(r"Service \d+ is not in project \d+\'s active search search results"),
+                },
+            ),
+            (
+                False,
+                True,
+                False,
+                True,
+                True,
+                None,
+                False,
+                404,
+                {
+                    "error": AnyStringMatching(r"Project \d+ doesn't have an active search"),
+                },
+            ),
+            (
+                True,
+                False,
+                True,
+                False,
+                False,
+                None,
+                False,
+                400,
+                {
+                    "error": AnyStringMatching(r"Project \d+ has not been locked"),
+                },
+            ),
+            (
+                True,
+                False,
+                False,
+                False,
+                False,
+                None,
+                False,
+                400,
+                {
+                    "error": AnyStringMatching(r"Project \d+ has not been locked"),
+                },
+            ),
+            (
+                True,
+                True,
+                False,
+                False,
+                True,
+                "cancelled",
+                True,
+                410,
+                {
+                    "error": AnyStringMatching(r"Project \d+ already has a completed process outcome: \d+"),
+                },
+            ),
+            (
+                True,
+                False,
+                True,
+                False,
+                True,
+                "awarded",
+                True,
+                410,
+                {
+                    "error": AnyStringMatching(r"Project \d+ already has a completed process outcome: \d+"),
+                },
+            ),
+        )
+    )
+    def test_most_paths(
```

Comment:
The unit tests doesn't test a unit. They test features. If a test fails on my test run called `test_most_paths` I have no idea what it tests or why it has failed. Part of having unit tests its to help us debug problems quickly.

Splitting the tests into units like audit event creation, paths for with and without active search, paths for successful and unsuccessful responses help with this and are descriptive. You can use the unit tests to determine what the code should be doing. You can often pinpoint exactly what has changed to trigger an error.

Updating the code then finding what part of the test to change and what parameters need to change is difficult when tests are too long.

Reply:
I'd rename the test to `def test_direct_award_outcome_scenarios(...)`.